atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTable.java

```diff
@@ -157,12 +157,15 @@ private void dropTableInternal(String fullTableName, String shortTableName) {
     @Override
     public void truncate() {
         try {
-            executeIgnoringError("TRUNCATE TABLE " + oracleTableNameGetter.getInternalShortTableName(),
-                    ORACLE_NOT_EXISTS_ERROR);
-            executeIgnoringError("TRUNCATE TABLE " + oracleTableNameGetter.getInternalShortOverflowTableName(),
-                    ORACLE_NOT_EXISTS_ERROR);
-        } catch (TableMappingNotFoundException e) {
-            // If table does not exist, do nothing
+            conns.get().executeUnregisteredQuery("TRUNCATE TABLE " + oracleTableNameGetter.getInternalShortTableName());
+            executeIgnoringError(
+                    "TRUNCATE TABLE " + oracleTableNameGetter.getInternalShortOverflowTableName(),
+                    "ORA-00942");
```

Comment:
I'll make you a deal and replace this and the 10,000 other ORA-n catches with constants in another PR, since I've also wanted this for a long time in PG as well